src/app/code/community/Bugsnag/Notifier/Model/Observer.php

```diff
@@ -63,13 +63,15 @@ public function initBugsnag()
     private function addUserTab()
     {
         $customer = Mage::getSingleton('customer/session')->getCustomer();
-        $this->client->setUser(array(
-            'id' => $customer->getId(),
-            'email' => $customer->getEmail(),
-            'created_at' => $customer->getCreatedAt(),
-            'first_name' => $customer->getFirstname(),
-            'last_name' => $customer->getLastname()
-        ));
+        if ($customer instanceof Mage_Customer_Model_Customer) {
+            $this->client->setUser(array(
+                'id' => $customer->getId(),
```

Comment:
Maybe worth just checking if those methods exist and are callable, so we can call them even if it's not an instance of that class, and still get partial information.

Reply:
Given that the `getCustomer()` method of the Session object should only ever return a `Mage_Customer_Model_Customer` object (or one which extends it) or null, I'd suggest that it would more indicative of an issue in the system if the `instanceof` condition doesn't  match yet the methods did. Without knowing that the data being sent over is genuinely a customer, would this not make the information less reliable?